### ISSUE TITLE
serviceEnabled() whenever it returns true

### ISSUE THREAD

**Issue by deiviszi** [Lyokone/flutterlocation#304]
**Describe the bug**
When gps is disabled and I call the serviceEnabled() method, it will only return true.

**Expected behavior**
Expected return false.

**Tested on:**
 - Android, API Level 28, real device, xiaomi redmi note 7 and xiaomi mi 8.

**Comment by deiviszi** [comment]
Good morning, from what I saw the problem is happening with real devices. I saw that the FlutterLocation.java file has the function:
`public boolean checkServiceEnabled (final Result result)`
In line 321 (plugin version 2.5.2) the code looks like this:
`if (gps_enabled || network_enabled) {`
Here it worked when I switched to:
`if (gps_enabled && network_enabled) {`

I don't know if it is a valid change, but I would like to report what I did in order to work.

**Comment by Lyokone** [comment]
Hey !
The location can be determined without a GPS (even tho it will be less accurate), so serviceEnabled return false when location service have been disabled, not only GPS :)

**Comment by deiviszi** [comment]
Okay, yes it is possible to use the network when using only the coarse location. But in my case I intend to get the fine location that needs the gps connected, so I made this temporary change to solve my problem. On emulating devices even if I keep the network on and turn off the gps, the result is being expected, I checked other issues that other people had the same problem with Xiaomi brand devices.
Thank you very much for your attention.
